### PR TITLE
Fix reading older layers

### PR DESCRIPTION
**Pull request by odisseus**
Copy-pasted from PR #1310.

### CODE REVIEW

File: src/core/layer.scala [hunk 1]
```diff
@@ -322,10 +322,14 @@ object Layer extends Lens.Partial[Layer] {
...
       migrate((version match {
+        case 7 =>
+          Try(ogdl.set(repos = ogdl.repos.map { repo =>
+            repo.set(remote = repo.repo)
+          })).getOrElse(ogdl)
         case 6 =>
           // FIXME: This is a lazy solution
           Try(ogdl.set(repos = ogdl.repos.map { repo =>
-            repo.set(branch = repo.track)
+            repo.set(branch = repo.track, remote = repo.repo)
```
Comment (propensive) [inline src/core/layer.scala]: This line change isn't required. Each update should be applied incrementally in turn.

File: src/core/layer.scala [hunk 2]
```diff
@@ -322,10 +322,14 @@ object Layer extends Lens.Partial[Layer] {
     if(version < CurrentVersion) {
       log.note(msg"Migrating layer file from version $version to ${version + 1}")
       migrate((version match {
+        case 7 =>
```
Comment (propensive) [inline src/core/layer.scala:325]: This should be close to identical to the changes in #1310.

**Review by propensive — APPROVED** [review]

File: src/core/layer.scala (hunk 2 above)
Comment (odisseus) [reply src/core/layer.scala:325]: Yes, that's where i took these changes.

File: src/core/layer.scala (hunk 1 above)
Comment (odisseus) [reply src/core/layer.scala]: Reverted

**Pull request merged**